otter/convergence/steps.py

```diff
@@ -93,12 +93,20 @@ class DeleteServer(object):
 
     def as_effect(self):
         """Produce a :obj:`Effect` to delete a server."""
-        return service_request(
+        eff = service_request(
             ServiceType.CLOUD_SERVERS,
             'DELETE',
             append_segments('servers', self.server_id),
             success_pred=has_code(204))
 
+        def report_success(result):
```

Comment:
Can this section be written as a decorator and `as_effect` can be decorated with that since most of the functions will have similar behavior?